### PR TITLE
add simple test to check plaintext lengths

### PR DESCRIPTION
**Pull request by tomato42**
### Description

Add test to verify if server is able to send and receive all valid sizes of ApplicationData

### Motivation and Context

https://github.com/openssl/openssl/issues/15706
https://github.com/tlsfuzzer/tlslite-ng/pull/480

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [ ] the changes are also reflected in documentation and code comments
- [x] all new and existing tests pass (see CI results)
- [ ] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [x] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [ ] new and modified scripts were ran against popular TLS implementations:
  - [x] OpenSSL
  - [ ] NSS
  - [ ] GnuTLS
- [x] required version of tlslite-ng updated in requirements.txt and README.md

---
This change is [Reviewable](https://reviewable.io/reviews/tlsfuzzer/tlsfuzzer/762)

### CODE REVIEW

**Comment by wmgl5593** [comment]
> ### Description
> ### Motivation and Context
> ### Checklist
> * [ ]  I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
> * [ ]  the changes are also reflected in documentation and code comments
> * [ ]  all new and existing tests pass (see CI results)
> * [ ]  [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
> * [ ]  new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
> * [ ]  new and modified scripts were ran against popular TLS implementations:
>   
>   * [ ]  OpenSSL
>   * [ ]  NSS
>   * [ ]  GnuTLS
> * [ ]  required version of tlslite-ng updated in requirements.txt and README.md
> 
> This change is [Reviewable](https://reviewable.io/reviews/tlsfuzzer/tlsfuzzer/762)

File: scripts/test-tls13-lengths.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    print("                names and not all of them, e.g \"sanity\"")
+    print(" -e probe-name  exclude the probe from the list of the ones run")
+    print("                may be specified multiple times")
+    print(" -x probe-name  expect the probe to fail. When such probe passes despite being marked like this")
+    print("                it will be reported in the test summary and the whole script will fail.")
+    print("                May be specified multiple times.")
+    print(" -X message     expect the `message` substring in exception raised during")
+    print("                execution of preceding expected failure probe")
+    print("                usage: [-x probe-name] [-X exception], order is compulsory!")
+    print(" -n num         run 'num' or all(if 0) tests instead of default(all)")
```
Comment (mahavrila) [inline scripts/test-tls13-lengths.py:50]: Is the 'default(all)' correct? Isn't it 1000 based on num_limit defined at line 61?

File: scripts/test-tls13-lengths.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    print("                (\"sanity\" tests are always executed)")
+    print(" -C cipher      ciphersuite to use for the connection")
```
Comment (mahavrila) [inline scripts/test-tls13-lengths.py:52]: This help section usually mentions default value. Is it intended that no default cipher is mentioned?

File: scripts/test-lengths.py [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+    print(" -d             use TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256")
+    print("                instead of TLS_RSA_WITH_AES_128_CBC_SHA256")
+    print(" -C cipher      ciphersuite to use for the connection")
+    print(" --extra-exts   send also supported_groups, signature_algorithms,")
+    print("                and signature_algorithms_cert extensions in Client")
+    print("                Hello. Default for DHE and ECDHE ciphers")
+    print(" -t timeout     Connection and server reply timeout, in seconds.")
+    print("                5 seconds by default.")
+    print(" --etm          Advertise and expect encrypt_then_mac extension")
+    print(" --size-limit num Send a max_fragment_length extension advertising")
```
Comment (mahavrila) [inline scripts/test-lengths.py:59]: Please check this line.

File: scripts/test-lengths.py [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+    print("                names and not all of them, e.g \"sanity\"")
+    print(" -e probe-name  exclude the probe from the list of the ones run")
+    print("                may be specified multiple times")
+    print(" -x probe-name  expect the probe to fail. When such probe passes despite being marked like this")
+    print("                it will be reported in the test summary and the whole script will fail.")
+    print("                May be specified multiple times.")
+    print(" -X message     expect the `message` substring in exception raised during")
+    print("                execution of preceding expected failure probe")
+    print("                usage: [-x probe-name] [-X exception], order is compulsory!")
+    print(" -n num         run 'num' or all(if 0) tests instead of default(all)")
```
Comment (mahavrila) [inline scripts/test-lengths.py:48]: Is the default really 'all'? Isn't it 1000?

File: scripts/test-lengths.py [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+    print("                (\"sanity\" tests are always executed)")
+    print(" -d             use TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256")
+    print("                instead of TLS_RSA_WITH_AES_128_CBC_SHA256")
+    print(" -C cipher      ciphersuite to use for the connection")
```
Comment (mahavrila) [inline scripts/test-lengths.py:52]: Is it intended that no default cipher is mentioned?

**Review by mahavrila — COMMENTED** [review]
I spotted only a few potential inaccuracies in help messages. Please check comments and consider them. Otherwise looks good to me.

File: scripts/test-tls13-lengths.py (hunk 1 above)
Comment (tomato42) [reply scripts/test-tls13-lengths.py:50]: right, fixed

File: scripts/test-tls13-lengths.py (hunk 2 above)
Comment (tomato42) [reply scripts/test-tls13-lengths.py:52]: true, good practice, added

File: scripts/test-lengths.py (hunk 4 above)
Comment (tomato42) [reply scripts/test-lengths.py:48]: right, fixed

File: scripts/test-lengths.py (hunk 3 above)
Comment (tomato42) [reply scripts/test-lengths.py:59]: I don't think there's an issue here, leaving as-is

Comment (mahavrila) [reply scripts/test-lengths.py:59]: Pardon, I misunderstood this line and there is actually no issue.

**Review by mahavrila — APPROVED** [review]
Looks good to me.

**Pull request merged**